3-enrich/scala-common-enrich/src/test/scala/com.snowplowanalytics.snowplow.enrich.common/adapters/registry/AdapterSpec.scala

```diff
@@ -134,4 +135,13 @@ class AdapterSpec extends Specification with DataTables with ValidationMatchers
     val expected = NonEmptyList(RawEvent(Shared.api, Map("tv" -> "com.adapter-v1", "e" -> "ue", "p" -> "srv"), Shared.contentType.some, Shared.cljSource, Shared.context),RawEvent(Shared.api, Map("tv" -> "com.adapter-v1", "e" -> "ue", "p" -> "srv"), Shared.contentType.some, Shared.cljSource, Shared.context),RawEvent(Shared.api, Map("tv" -> "com.adapter-v1", "e" -> "ue", "p" -> "srv"), Shared.contentType.some, Shared.cljSource, Shared.context))
     BaseAdapter.rawEventsListProcessor(validatedRawEventsList) must beSuccessful(expected)
   }
+
+  def e9 =
+      "SPEC NAME"                       || "JSON"                                             | "EXPECTED OUTPUT"                                                                                                              |
+      "Change one value"                !! """{"ts":1415709559}"""                            ! JObject(List(("ts",JString("2014-11-11T12:39:19.000Z"))))                                                                      |
+      "Change multiple values"          !! """{"ts":1415709559,"ts":1415700000}"""            ! JObject(List(("ts",JString("2014-11-11T12:39:19.000Z")),("ts",JString("2014-11-11T10:00:00.000Z"))))                           |
+      "Change nested values"            !! """{"ts":1415709559,"nested":{"ts":1415700000}}""" ! JObject(List(("ts",JString("2014-11-11T12:39:19.000Z")),("nested",JObject(List(("ts",JString("2014-11-11T10:00:00.000Z"))))))) |
+      "JStrings should also be changed" !! """{"ts":"1415709559"}"""                          ! JObject(List(("ts",JString("2014-11-11T12:39:19.000Z"))))                                                                                    |> {
+      (_, json, expected) => MandrillAdapter.cleanupJsonEventValues(parse(json), None, "ts") mustEqual expected
```

Comment:
It's not, no! I'm going to push what I've got as it fixes this, however its still WIP and contains a few eyebrow raisers still